### PR TITLE
allow unload to be filtered with where clause

### PR DESCRIPTION
**Pull request by andharris**
We should allow the `unload` method to be filtered by a user provided SQL where clause.

### CODE REVIEW

File: shiftmanager/mixins/s3.py [hunk 1]
```diff
@@ -464,13 +467,18 @@ def unload_table_to_s3(self, bucket, keypath, table, col_str='*',
...
             cols = col_str
 
+        select = "SELECT {col_str} FROM {table} ".format(
+            col_str=cols, table=table)
+        if where:
+            select += where
+
         statement = """
-        UNLOAD ($$SELECT {col_str} FROM {table}$$)
+        UNLOAD ($$ {select} $$)
```
Comment (jklukas) [inline shiftmanager/mixins/s3.py]: Unnecessary spaces here around {select}

File: shiftmanager/mixins/s3.py [hunk 2]
```diff
@@ -464,13 +467,18 @@ def unload_table_to_s3(self, bucket, keypath, table, col_str='*',
         else:
             cols = col_str
 
+        select = "SELECT {col_str} FROM {table} ".format(
+            col_str=cols, table=table)
+        if where:
```
Comment (jklukas) [inline shiftmanager/mixins/s3.py]: When the default is None, I expect this to be `if where is not None`

File: shiftmanager/mixins/s3.py [hunk 3]
```diff
@@ -435,6 +435,9 @@ def unload_table_to_s3(self, bucket, keypath, table, col_str='*',
         col_str : str
             Comma separated string of columns to unload
             Defaults to '*'
+        where : str
+            SQL where clause string to filter select statement in unload
+            Defaults to None
```
Comment (jklukas) [inline shiftmanager/mixins/s3.py]: Defaults to None, meaning no WHERE clause is applied

File: shiftmanager/mixins/s3.py (hunk 1 above)
Comment (andharris) [reply shiftmanager/mixins/s3.py]: I found this a little easier to read but happy to remove

File: shiftmanager/mixins/s3.py (hunk 2 above)
Comment (andharris) [reply shiftmanager/mixins/s3.py]: 👍

File: shiftmanager/mixins/s3.py (hunk 3 above)
Comment (andharris) [reply shiftmanager/mixins/s3.py]: good call

**Pull request merged**